lib/linalg/BasisGenerator.cpp

```diff
@@ -132,6 +132,7 @@ BasisGenerator::takeSample(
     // Check that u_in is not non-zero.
     Vector u_vec(u_in, getDim(), true);
     if (u_vec.norm() == 0.0) {
+        printf("BasisGenerator::takeSample - trivial samples are not allowed!\n");
```

Comment:
This warning message is helpful, but it could be more informative. First, it may not be clear whether a serious error occurred, which would cause a failure. In this case, it is simply a warning that the sample was skipped. Maybe something like this would be clearer:

std::cout << "WARNING: BasisGenerator::takeSample skipped trivial sample at time " << time << std::endl;

Reply:
Revised the error message, reflecting this comment.